impl/http/src/main/java/io/serverlessworkflow/impl/executors/http/OAuth2AuthProvider.java

```diff
@@ -16,24 +16,53 @@
 package io.serverlessworkflow.impl.executors.http;
 
 import io.serverlessworkflow.api.types.OAuth2AuthenticationPolicy;
+import io.serverlessworkflow.api.types.Oauth2;
 import io.serverlessworkflow.api.types.Workflow;
+import io.serverlessworkflow.http.jwt.JWT;
 import io.serverlessworkflow.impl.TaskContext;
 import io.serverlessworkflow.impl.WorkflowApplication;
 import io.serverlessworkflow.impl.WorkflowContext;
 import io.serverlessworkflow.impl.WorkflowModel;
+import io.serverlessworkflow.impl.executors.http.oauth.OAuthRequestBuilder;
+import jakarta.ws.rs.client.Invocation;
 import jakarta.ws.rs.client.Invocation.Builder;
 
 public class OAuth2AuthProvider implements AuthProvider {
 
+  private Oauth2 oauth2;
+
+  private final WorkflowApplication workflowApplication;
```

Comment:
Rather than having these two private members, replace them by private OAuthRequestBuilder requestBuilder and initilize them passing the WorkflowApplication and OAuth2. 
Then, in prerequest you call build over this new member.